### PR TITLE
Add Eclipse suggested CODE_OF_CONDUCT.md

### PR DESCRIPTION
**Pull request by pshipton**
Eclipse recommends to have this file as per the yearly release review.
https://gitlab.eclipse.org/eclipsefdn/emo-team/emo/-/issues/106

Copied the file from
https://github.com/eclipse/.github/blob/master/CODE_OF_CONDUCT.md

### CODE REVIEW

**Comment by DanHeidinga** [comment]
Looks reasonable to me and I would expect that we're already operating in this fashion.  

I'm unclear on the `responsibilities` in:
> Project committers and leaders have the right and responsibility to remove, edit, or reject comments, commits, code, wiki edits, issues, and other contributions that are not aligned to this Code of Conduct, or to ban temporarily or permanently any contributor for other behaviors that they deem inappropriate, threatening, offensive, or harmful.

Most of this is straightforward but as we've never had to go down the `ban temporarily or permanently any contributor ` path, I have no idea if we have the capabilities to actually do that.  I'm hesitant to sign up to responsibility until we're clear we can actually take the required action.

Does anyone know if there is a way to ban a contributor from github?  (...goes off to search....)

**Comment by DanHeidinga** [comment]
> Does anyone know if there is a way to ban a contributor from github? (...goes off to search....)

Only Organization owners can block a user [1] on github so we can't actually enforce the responsibility we're taking on here.  This might need some slight rewording to state that the Foundation will..... or they'll need to grant us further permissions on the org.


[1] https://docs.github.com/en/communities/maintaining-your-safety-on-github/blocking-a-user-from-your-organization

**Comment by DanHeidinga** [comment]
I hate promising actions I can't take ie: "to ban temporarily or permanently any contributor for....".  I'm less worried about committers and more concerned about with if we ever have to deal with a trolls =)

**Comment by mstoodle** [comment]
Note we already mention the code of conduct in our README, and although we don't explicitly state that the project follows it, it is implied. Under the "How do I contribute?" section: `Since we are an Eclipse Foundation project, each contributor needs to sign an Eclipse Contributor Agreement. The Eclipse Foundation operates under the Eclipse Code of Conduct to promote fairness, openness, and inclusion.`

How about we take a committer vote, and in that issue, Dan can propose some words to scope the ban requirement to something we can more comfortably commit to? As in, we commit to follow this code with the following understanding...

**Comment by pshipton** [comment]
@DanHeidinga given https://github.com/eclipse-openj9/openj9/pull/13637#issuecomment-937857381 and the lack of response to https://github.com/eclipse/.github/issues/13, are you will to accept the code of conduct as-is? If so we can start making sure all the other committers are aware of this and see about merging it.

**Comment by DanHeidinga** [comment]
I've pinged Wayne on eclipse/.github#13 and expect we'll get an answer in the next week or two

**Comment by waynebeaton** [comment]
Sorry, I don't monitor that repository's inbox.

As I stated previously, my preference is that we not start creating minor variations of the code of conduct unless absolutely necessary. I don't see why a change would be necessary in this case.

**Comment by pshipton** [comment]
@DanHeidinga are you still going to block this due to the wording, which Eclipse doesn't see the need to change, or can we proceed trying to get all committers to approve? It's time for our yearly release review and it's recommended to have the code of conduct file.

**Comment by DanHeidinga** [comment]
I continue to be unimpressed by the Foundation's response to the issue of responsibility without authority (and no their proposed "mother may I" model doesn't cut it).  Unfortunately, they don't seem to be in any rush to resolve the issue.

That said, I don't see anything in the code of conduct that we aren't already trying to live out in the project and I don't want to have this wait around.

Let's proceed.

**Review by dmitripivkine — APPROVED** [review]

**Review by dsouzai — APPROVED** [review]

**Review by ymanton — APPROVED** [review]

**Review by AdamBrousseau — APPROVED** [review]

**Review by jdmpapin — APPROVED** [review]

**Review by hzongaro — APPROVED** [review]

**Review by knn-k — APPROVED** [review]

**Review by 0xdaryl — APPROVED** [review]

**Review by AdamBrousseau — COMMENTED** [review]
I suppose if this file doesn't need a copyright header, we should add it to the `.copyrightignore` file.

**Comment by keithc-ca** [comment]
> I suppose if this file doesn't need a copyright header, we should add it to the `.copyrightignore` file.

We could, but I would hope this file doesn't change much; it may be easier for committers to just ignore the copyright checker for those few times it does change.

**Review by mstoodle — APPROVED** [review]

**Review by llxia — APPROVED** [review]

**Review by mpirvu — APPROVED** [review]

**Review by gita-omr — APPROVED** [review]

**Review by smlambert — APPROVED** [review]

**Review by amicic — APPROVED** [review]

**Review by joransiu — APPROVED** [review]

**Review by vijaysun-omr — APPROVED** [review]

**Review by gacholio — APPROVED** [review]

**Review by tajila — APPROVED** [review]

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by contacting the Staff at codeofconduct@eclipse.org. All complaints will be reviewed and investigated and will result in a response that is deemed necessary and appropriate to the circumstances. The Staff is obligated to maintain confidentiality with regard to the reporter of an incident. Further details of specific enforcement policies may be posted separately.
+
+Project committers or leaders who do not follow the Code of Conduct in good faith may face temporary or permanent repercussions as determined by the Staff.
+
+## Attribution
+
+This Code of Conduct is adapted from the [Contributor Covenant](https://www.contributor-covenant.org) , version 1.4, available at [https://www.contributor-covenant.org/version/1/4/code-of-conduct.html](https://www.contributor-covenant.org/version/1/4/code-of-conduct/)
```
Comment (babsingh) [inline CODE_OF_CONDUCT.md:46]:
```suggestion
This Code of Conduct is adapted from the [Contributor Covenant](https://www.contributor-covenant.org), version 1.4, available at [https://www.contributor-covenant.org/version/1/4/code-of-conduct.html](https://www.contributor-covenant.org/version/1/4/code-of-conduct/).
```

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+## Our Standards
+
+Examples of behavior that contributes to creating a positive environment include:
+
+*   Using welcoming and inclusive language
+*   Being respectful of differing viewpoints and experiences
+*   Gracefully accepting constructive criticism
+*   Focusing on what is best for the community
+*   Showing empathy towards other community members
```
Comment (babsingh) [inline CODE_OF_CONDUCT.md:18]: also, can be full-stops.
```suggestion
*   Using welcoming and inclusive language;
*   Being respectful of differing viewpoints and experiences;
*   Gracefully accepting constructive criticism;
*   Focusing on what is best for the community; and
*   Showing empathy towards other community members.
```

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+Examples of unacceptable behavior by participants include:
+
+*   The use of sexualized language or imagery and unwelcome sexual attention or advances
+*   Trolling, insulting/derogatory comments, and personal or political attacks
+*   Public or private harassment
+*   Publishing others' private information, such as a physical or electronic address, without explicit permission
+*   Other conduct which could reasonably be considered inappropriate in a professional setting
```
Comment (babsingh) [inline CODE_OF_CONDUCT.md:26]: also, can be full-stops.
```suggestion
*   The use of sexualized language or imagery and unwelcome sexual attention or advances;
*   Trolling, insulting/derogatory comments, and personal or political attacks;
*   Public or private harassment;
*   Publishing others' private information, such as a physical or electronic address, without explicit permission; and
*   Other conduct which could reasonably be considered inappropriate in a professional setting.
```

**Review by babsingh — APPROVED** [review]
Approving, LGTM. Identified minor punctuation nitpicks.

**Review by doveye — APPROVED** [review]

**Comment by pshipton** [comment]
@fjeremic @jduimovich @charliegracie @pmhayward  if you want to sign off on this, if you ever plan to participate as an OpenJ9 committer again.

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (pshipton) [reply CODE_OF_CONDUCT.md:46]: As it's copied verbatim [from Eclipse](https://github.com/eclipse/.github/blob/master/CODE_OF_CONDUCT.md), I'm going to keep is as-is. The punctuation doesn't seem important enough to vary from the original, and make it harder to compare and pick up future changes.

**Comment by doveye** [comment]
Pretty sure pmhayward has no plans to contribute in the future

**Review by zl-wang — APPROVED** [review]

**Comment by pshipton** [comment]
@DanHeidinga all the regular committers have approved. There are some inactive committers who haven't responded (yet). Can this be merged now?

**Review by DanHeidinga — APPROVED** [review]

**Comment by DanHeidinga** [comment]
> @DanHeidinga all the regular committers have approved. There are some inactive committers who haven't responded (yet). Can this be merged now?

I'm good with merging at this point.

**Pull request merged**